gflownet/envs/crystals.py

```diff
@@ -325,10 +330,23 @@ def step(self, action_idx):
             return self.state, action_idx, valid
         # If action is eos, then perform eos
         else:
-            if len(self.state) < self.min_atoms:
+            if sum(self.state) < self.min_atoms:
                 valid = False
             else:
-                self.done = True
-                valid = True
-                self.n_actions += 1
+                nums_charges = [(num, self.oxidation_states[i]) for i, num in enumerate(self.state) if num > 0]
+                sum_diff_elem = []
+                for n, c in nums_charges:
+                    charges = []
+                    for c in itertools.product(c, repeat=n):
+                        sc = sorted(c)
+                        if sc not in charges:
+                            charges.append(sc)
+                    sum_diff_elem.append([sum(ci) for ci in charges])
```

Comment:
Complementary comment to Michał's: I think it would be worth having a separate function to check the charge neutrality, like `has_neutral_charge(state)`